File: Data Structures & Algorithms/top-k-elements-in-list/submission-4.py
class Solution:
    def topKFrequent(self, nums: List[int], k: int) -> List[int]:
        d = {}
        l = []
        for i in nums:
            d[i] = d.get(i, 0) + 1
        for _ in range(k):
            l.append(max(d, key=d.get))
            del d[max(d, key=d.get)]
        return l

File: Data Structures & Algorithms/buy-and-sell-crypto/submission-4.cpp
class Solution {
public:
    int maxProfit(vector<int>& prices) {
        int ret = 0;
        int l = 0;
        for (int r = 0; r < prices.size(); ++r) {
            ret = max(ret, prices.at(r) - prices.at(l));
            if (prices.at(l) > prices.at(r)) l = r;
        }
        return ret;
    }
};
